### PR TITLE
links for map, website, gmail implemented

### PR DESCRIPTION
**Pull request by nirali25parekh**
Changes in this pull request:
In Single Project Details page:
- Link to respective locations in google maps
- Link to gmail with repsective receipent
- Link to its website
(shows 'unavailable' if any field is missing in the database)

### CODE REVIEW

File: src/features/public/Donations/screens/ProjectDetails.tsx [hunk 1]
```diff
@@ -18,23 +18,26 @@ interface Props {
...
 
 function ProjectDetails({ project }: Props): ReactElement {
+
+
   const [rating, setRating] = React.useState<number | null>(2);
 
   const progressPercentage =
     (project.countPlanted / project.countTarget) * 100 + '%';
   const ImageSource = project.image
     ? getImageUrl('project', 'large', project.image)
```
Comment (sagararyal) [inline src/features/public/Donations/screens/ProjectDetails.tsx:28]: – image medium should suffice for general purpose.

large can be loaded for retina image. Not sure if next has built in feature, but css norm is: to supply srcset.
eg:
``

File: src/features/public/Donations/components/projectDetails/ProjectContactDetails.tsx [hunk 2]
```diff
@@ -1,31 +1,40 @@
...
+  return (
+    <div className={styles.projectMoreInfo}>
+      <div className={styles.infoTitle}>Contact Details</div>
+      {contactDetails.map((contactDetails) => (
+        <div key={contactDetails.id}>
+          <div className={styles.infoText + ' ' + styles.contactDetailsRow}>
+            {contactDetails.icon}
+
+            {contactDetails.link ? (
+              <span style={{ marginLeft: '20px', flexGrow: 1 }}>
```
Comment (sagararyal) [inline src/features/public/Donations/components/projectDetails/ProjectContactDetails.tsx:23]: 16px ~ 1rem

File: src/features/public/Donations/screens/ProjectDetails.tsx [hunk 3]
```diff
@@ -18,23 +18,26 @@ interface Props {
...
     {
       id: 3,
       icon: <Location />,
-      text: '303 W Foothill Blvd, Unit 13 Glendora, CA 91741, USA',
-      link: '',
+      text: project.tpo.address ? project.tpo.address: 'unavailable',
+      link: project.coordinates ? `https://maps.google.com/?q=${project.coordinates.lat},${project.coordinates.lon}` : null,
     },
-    { id: 4, icon: <Email />, text: 'projects@edenprojects.org', link: '' },
+    { id: 4, icon: <Email />, text: project.tpo.email ? project.tpo.email : 'unavailable', link: project.tpo.email ? `https://mail.google.com/mail/?view=cm&fs=1&tf=1&to=${project.tpo.email}` : null },
```
Comment (sagararyal) [inline src/features/public/Donations/screens/ProjectDetails.tsx:40]: The link would be `mailto:email@example.com` instead of a gmail link.

**Comment by harshvitra** [comment]
Fixes #22 #23 #24

**Pull request merged**